resolver/resolver_test.go

```diff
@@ -417,27 +417,68 @@ func TestMultiError(t *testing.T) {
 	}
 }
 
-func TestTruncate(t *testing.T) {
-	tm := *newTruncated()
-	if !tm.Truncated {
+func TestTruncateSetTruncateBit(t *testing.T) {
+	testTruncateSetTruncateBit(t, dns.MinMsgSize)
+}
+
+func TestTruncateEdns0SetTruncateBit(t *testing.T) {
+	testTruncateSetTruncateBit(t, 4096)
+}
+
+func TestTruncateNoSetTruncateBit(t *testing.T) {
+	testTruncateNoSetTruncateBit(t, dns.MinMsgSize)
+}
+
+func TestTruncateEdns0NoSetTruncateBit(t *testing.T) {
+	testTruncateNoSetTruncateBit(t, 4096)
+}
+
+func testTruncateSetTruncateBit(t *testing.T, maxSize uint16) {
+	orig := *newMessage(maxSize)
```

Comment:
I don't think there's any point in this `orig` `tmp` thing. They aren't treated separately later on anyways.